nemo/collections/tts/data/text_to_speech_dataset.py

```diff
@@ -606,7 +606,12 @@ def __getitem__(self, index):
             align_prior = torch.tensor(align_prior, dtype=torch.float32)
             example["align_prior"] = align_prior
 
-        example['raw_text'] = data.text
+        if "original_text" in data.manifest_entry:
+            # For Manifests in which text field is IPA.
+            example['raw_text'] = data.manifest_entry['original_text']
+        else:
+            example['raw_text'] = data.text
```

Comment:
We should refactor this since this is a non-standard change to our manifest structure and dataset class

Reply:
What is raw_text being used for?

Reply:
@paarthneekhara  `raw_texts` and  is used in preference optimization for CER and WER computation in DPO data generation and GRPO. This is when `text` field contains the IPA string (used as input to the model), but we need the original text for computing CER and WER of the generated audio.

@blisc by refactoring do you mean I should add the original_text field in DatasetSample?

```
@dataclass
class DatasetSample:
    dataset_name: str
    manifest_entry: Dict[str, Any]
    audio_dir: Path
    feature_dir: Path
    text: str
    original_text: str = None # Used for CER/WER computation when text field contains phonemized/processed text
    speaker: str
    speaker_index: int = None
    tokenizer_names: List[str] = None
   ```

Reply:
> @paarthneekhara raw_texts and is used in preference optimization for CER and WER computation in DPO data generation and GRPO. This is when text field contains the IPA string (used as input to the model), but we need the original text for computing CER and WER of the generated audio.

This should be explained in our code. Ideally, our dataset or model would have this docstringed somewhere. For this PR, we should add this explanation here.